packages/meditrak-server/src/apiV2/import/importSurveyResponses/importSurveyResponses.js

```diff
@@ -82,15 +99,56 @@ export async function importSurveyResponses(req, res) {
       const [tabName, sheet] = surveySheets;
       const deletedResponseIds = new Set();
       const questionIds = [];
+      let survey;
 
       // extract response ids and set up update batcher
       const { maxColumnIndex, maxRowIndex } = getMaxRowColumnIndex(sheet);
       const minSurveyResponseIndex = INFO_COLUMN_HEADERS.length;
       const surveyResponseIds = [];
+      const isGeneratedIdByColumnIndex = [];
+      const existingResponseDataByColumnIndex = [];
```

Comment:
Local maps like the above are useful for caching data and avoiding redundant queries. At the same time, they add complexity to the code.

I think it would be beneficial to implement [NOT-894: Decouple input parsing and core logic in survey response import](https://linear.app/bes/issue/NOT-894/decouple-input-parsing-and-core-logic-in-survey-response-import). The suggested refactoring could help reduce complexity and improve performance at the same time, by following a multi-step approach like the following:

1. Validate/sanitise/parse spreadsheet data
2. Use the parsed data to fetch required records in an optimised way (eg all entities at once). At the moment, this is a bit hard since we parse + fetch as we go
3. Use the parsed data + fetched records to mutate survey response records